Engines/Wine/Verbs/FAudio/script.js

```diff
@@ -0,0 +1,43 @@
+include(["engines", "wine", "engine", "object"]);
+include(["utils", "functions", "net", "resource"]);
+include(["utils", "functions", "filesystem", "files"]);
+
+/**
+* Verb to install FAudio
+* see: https://github.com/Kron4ek/FAudio-Builds
+* @returns {Wine} Wine object
+*/
+Wine.prototype.faudio = function () {
+    var setupFile = new Resource()
+        .wizard(this.wizard())
+        .url("https://github.com/Kron4ek/FAudio-Builds/releases/download/19.01/faudio-19.01.tar.xz")
+        .name("faudio-19.01.tar.xz")
+        .get();
+
+    new Extractor()
+        .wizard(this.wizard())
+        .archive(setupFile)
+        .to(this.prefixDirectory() + "/FAudio/")
+        .extract();
+
+    this.runInsidePrefix("/FAudio/faudio-19.01/wine_setup_faudio.sh")
```

Comment:
I don't like to do this in the scripts.
@plata @madoar Any advice?

Reply:
This is just because the .tar.xz provide a way to install the dlls and to de overrides through a .sh, but I know this will not work (that's why I asked).

I can do everything in the script if you prefer.

Reply:
There is 38 dll to copy though, so I would prefer to use the .sh ^^

Reply:
ls returns this:`[Ljava.lang.String;@7fecc4b2`, so the forEach fails. Any idea ?

Reply:
Interesting. We need to fix it so that it returns a real list

Reply:
@qparis I agree, we shouldn't call shell scripts from inside our scripts, at least if it is not absolutely necessary.

Reply:
@ImperatorS79 can you check other places where `ls` is used? There should be a hint somewhere how it must be done.

Reply:
Well 
```js
/**
* lists files and directories
* @param {string} directoryPath directory path
* @returns {string[]} list of files and directories
*/
function ls(directoryPath) { // eslint-disable-line no-unused-vars
    return fileUtilities.ls(new java.io.File(directoryPath));
}
```
I do not understand why it does not work

Reply:
I mean a script using `ls`, not the implementation of it.

Reply:
In the function _installWinePackage
```js
var files = ls(localDirectory);
        if (files.length == 1) {
            // probably the archive contained a folder (e.g. for Lutris Wine version)
            // link folders so Phoenicis can find them
            var extractedDir = files[0];

            var forEach = Array.prototype.forEach;
            forEach.call(ls(localDirectory + "/" + extractedDir), function (folder) {
                lns(localDirectory + "/" + extractedDir + "/" + folder, localDirectory + "/" + folder);
            }
            );
}
```

Reply:
So I do not see why is does not work

Reply:
Yep something like this. Should also work for you or not?

Reply:
@plata I cannot call this.system32directory(), this.override_dll inside the foreach.call.

Reply:
OK, I used self=this ^^.